collector/info_schema_innodb_metrics.go

```diff
@@ -35,6 +53,21 @@ func ScrapeInnodbMetrics(db *sql.DB, ch chan<- prometheus.Metric) error {
 		); err != nil {
 			return err
 		}
+		// Special handling of the "buffer_page_io" subsystem.
+		if subsystem == "buffer_page_io" {
+			match := bufferPageRE.FindStringSubmatch(name)
+			switch match[1] {
```

Comment:
Check before this that `len(match)` is really at least 3 (and maybe log an error if not?). Otherwise the whole exporter will just crash in case this is ever not true.